#<h1><center>**********INDEX**********</center></h1><br>


1. [Pre requisites](#p1)<br>
1 1. [Important URL's for quick reference](#s1)<br>
1 2. [Installing alll required Packages ](#s2)<br>
1 3. [Importing - Packages/Modules](#s3)<br>
1 3 1.[Importing Data from the Google Drive](#s31)<br>
1 4. [Connecting to the Database](#s4)<br>
1 5. [Fetch Data from the Tables](#s5)<br>
1 6. [Find the Count of Rows from each Table](#s6)<br>
1 7. [Extracting data from respective Tables](#s7)<br>
1 7 1. [Table - Gender_Age_Train](#s8)<br>
1 7 2. [Table - Phone_Brand_Device](#s9)<br>
1 8. [Copy Data from the Google Drive](#s10)<br>
2. [Working on Events_data](#s11)<br>
3. [Working on Gender_Age_Train](#s12)<br>
4. [Working on Phone_Brand_Device_Model](#s13)<br>
5. [Observations and Handling the data](#s14)<br>


# 1 Pre requsites <a name = p1></a>

##1 1 Important URL's for quick reference <a name = "s1"></a>

https://www.w3schools.com/python/python_mysql_select.asp <br>
https://www.datacamp.com/community/tutorials/mysql-python <br>
https://www.databasejournal.com/features/mysql/article.php/3904531/The-10-Most-Common-MySQL-Queries.htm <br>



##1 2 Installing all required packages <a name = "s2"></a>
<br>1. mysql.connector
<br>2. sqlalchemy
<br>3. psycopg2-binary

In [42]:
!pip install mysql.connector
!pip install sqlalchemy
!pip install psycopg2-binary
!pip install pymysql

     |████████████████████████████████| 11.9MB 259kB/s 
  Created wheel for mysql.connector: filename=mysql_connector-2.2.9-cp36-cp36m-linux_x86_64.whl size=247949 sha256=4498af19a493abe8cab0ee2dbbf3bd9630b79f71a8debfa38dc201661fe625c0
  Stored in directory: /root/.cache/pip/wheels/8c/83/a1/f8b6d4bb1bd6208bbde1608bbfa7557504bed9eaf2ecf8c175
Successfully built mysql.connector
     |████████████████████████████████| 2.9MB 2.8MB/s 
     |████████████████████████████████| 51kB 1.6MB/s 


##1 3 Importing all required Packages and Modules <a name = "s3"></a>

In [43]:
# Importing mysql Packages

import mysql.connector as mysql
from sqlalchemy import create_engine

print("Import completed")

Import completed


In [44]:
#Importing Pandas 

import pandas as pd
import numpy as np
print("Import completed")

Import completed


### 1 3 1 Import data from Google Drive <a name = "s31"></a>

In [45]:
# To connect to Google Drive : 

from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## 1 4 Connecting the Database<a name = "s4"></a>

In [46]:
db = mysql.connect(
    host = "cpanel.insaid.co",
    user = "student",
    passwd = "student",
    database = "Capstone1"
)
print("Connection successful - Details of the connected DB : ", db)

Connection successful - Details of the connected DB :  <mysql.connector.connection.MySQLConnection object at 0x7ff94caeeda0>


## 1 5 Fetch details of Tables <a name = "s5"></a>

In [47]:
cursor = db.cursor()
## getting all the tables which are present in 'db' database
cursor.execute("SHOW TABLES")
tables = cursor.fetchall() ## it returns list of tables present in the database
## showing all the tables one by one
print("List of Tables under the database : student")
for table in tables:
    print(table)

List of Tables under the database : student
('events_data',)
('gender_age_train',)
('phone_brand_device_model',)


## 1 6 Find the Count of Rows from each Table<a name = "s6"></a>

In [48]:
cursor = db.cursor()
cursor.execute("select count(*) from events_data")
tables = cursor.fetchall() ## it returns list of tables present in the database
## showing all the tables one by one
for table in tables:
    print("Table events_data has ", table, " rows")

cursor.execute("select count(*) from gender_age_train")
tables = cursor.fetchall() ## it returns list of tables present in the database
## showing all the tables one by one
for table in tables:
    print("Table gender_age_train has ", table, " rows")

cursor.execute("select count(*) from phone_brand_device_model")
tables = cursor.fetchall() ## it returns list of tables present in the database
## showing all the tables one by one
for table in tables:
    print("Table phone_brand_device_model has ", table, " rows")

Table events_data has  (3252950,)  rows
Table gender_age_train has  (74645,)  rows
Table phone_brand_device_model has  (87726,)  rows


##1 7 Extracting rows from Tables<a name = "s7"></a>

### 1 7 1 Table - Gener_Age_Train<a name = "s8"></a>

In [0]:
engine = create_engine('mysql+pymysql://student:student@cpanel.insaid.co:3306/Capstone1')
df = pd.read_sql_table("gender_age_train", engine)
df.to_csv('gender_age_train.csv') 

###1 7 2 Table - Phone_Brand_Device_Model<a name = "s9"></a>


In [0]:
engine = create_engine('mysql+pymysql://student:student@cpanel.insaid.co:3306/Capstone1')
df = pd.read_sql_table("phone_brand_device_model", engine)
df.to_csv('phone_brand_device_model.csv') 

## 1 8 Copy data from the Google Drive<a name = "s10"></a>

In [51]:
!ls drive/'My Drive'/1014capstone/
!mkdir -p data
!cp -R drive/'My Drive'/1014capstone/* data/

events_data.csv  gender_age_train.csv  phone_brand_device_model.csv


# 2 Working on Events_data<a name = "s11"></a>

## 2 1 Read Events_data to a DataFrame

In [0]:
dt_evnts = pd.read_csv('/content/data/events_data.csv')

##2 2 Shape of the Events_data



In [53]:
dt_evnts.shape

(3252950, 7)

## 2 3 Display of first 10 rows

In [54]:
dt_evnts.head(10)

event_id     device_id            timestamp  ...  latitude   city  state
0       1.0  2.918269e+16  2016-05-01 00:55:25  ...  28.68161  Delhi  Delhi
1    7104.0  2.918269e+16  2016-05-02 09:37:02  ...  28.68161  Delhi  Delhi
2   29661.0  2.918269e+16  2016-05-04 00:56:04  ...  28.68161  Delhi  Delhi
3   33133.0  2.918269e+16  2016-05-06 05:01:15  ...  28.68161  Delhi  Delhi
4   38980.0  2.918269e+16  2016-05-06 09:55:04  ...  28.68161  Delhi  Delhi
5   66398.0  2.918269e+16  2016-05-01 01:56:06  ...  28.68161  Delhi  Delhi
6   70894.0  2.918269e+16  2016-05-06 01:14:46  ...  28.68161  Delhi  Delhi
7   87185.0  2.918269e+16  2016-05-01 09:20:01  ...  28.68161  Delhi  Delhi
8   94644.0  2.918269e+16  2016-05-07 07:16:35  ...  28.68161  Delhi  Delhi
9  110018.0  2.918269e+16  2016-05-06 21:00:40  ...  28.68161  Delhi  Delhi

[10 rows x 7 columns]

##2 4 Columns of the Events_data


In [55]:
dt_evnts.columns

Index(['event_id', 'device_id', 'timestamp', 'longitude', 'latitude', 'city',
       'state'],
      dtype='object')

##2 5 Information of the Events_data

In [56]:
dt_evnts.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3252950 entries, 0 to 3252949
Data columns (total 7 columns):
event_id     float64
device_id    float64
timestamp    object
longitude    float64
latitude     float64
city         object
state        object
dtypes: float64(4), object(3)
memory usage: 173.7+ MB


##2 6 Description of the Events data


In [57]:
dt_evnts.describe()

event_id     device_id     longitude      latitude
count  3.252950e+06  3.252497e+06  3.252527e+06  3.252527e+06
mean   1.626476e+06  1.012200e+17  7.815868e+01  2.168851e+01
std    9.390459e+05  5.316758e+18  4.235570e+00  5.789111e+00
min    1.000000e+00 -9.222957e+18  1.256700e+01  8.190110e+00
25%    8.132382e+05 -4.540611e+18  7.583543e+01  1.780171e+01
50%    1.626476e+06  1.726820e+17  7.726814e+01  2.216454e+01
75%    2.439713e+06  4.861813e+18  8.031916e+01  2.868278e+01
max    3.252950e+06  9.222849e+18  9.545946e+01  4.187190e+01

In [58]:
dt_evnts.describe(include="all")

event_id     device_id  ...     city    state
count   3.252950e+06  3.252497e+06  ...  3252950  3252573
unique           NaN           NaN  ...      933       32
top              NaN           NaN  ...    Delhi    Delhi
freq             NaN           NaN  ...   744276   751733
mean    1.626476e+06  1.012200e+17  ...      NaN      NaN
std     9.390459e+05  5.316758e+18  ...      NaN      NaN
min     1.000000e+00 -9.222957e+18  ...      NaN      NaN
25%     8.132382e+05 -4.540611e+18  ...      NaN      NaN
50%     1.626476e+06  1.726820e+17  ...      NaN      NaN
75%     2.439713e+06  4.861813e+18  ...      NaN      NaN
max     3.252950e+06  9.222849e+18  ...      NaN      NaN

[11 rows x 7 columns]

##2 7 Counting Null of each column

In [59]:
dt_evnts.isnull().sum()

event_id       0
device_id    453
timestamp      0
longitude    423
latitude     423
city           0
state        377
dtype: int64

# 3 Working on Gender_Age_Train<a name = "s12"></a>

##3 1 Read GenderAgeTrain to a DataFrame

In [0]:
dt_gat = pd.read_csv('/content/data/gender_age_train.csv')

##3 2 Shape of the GAT data

In [61]:
dt_gat.shape

(74645, 5)

##3 3 Dispaly of first 10 rows

In [62]:
dt_gat.head(10)

Unnamed: 0            device_id gender  age   group
0           0 -8076087639492063270      M   35  M32-38
1           1 -2897161552818060146      M   35  M32-38
2           2 -8260683887967679142      M   35  M32-38
3           3 -4938849341048082022      M   30  M29-31
4           4   245133531816851882      M   30  M29-31
5           5 -1297074871525174196      F   24  F24-26
6           6   236877999787307864      M   36  M32-38
7           7 -8098239495777311881      M   38  M32-38
8           8   176515041953473526      M   33  M32-38
9           9  1596610250680140042      F   36  F33-42

##3 4 Columns of the GAT

In [63]:
dt_gat.columns

Index(['Unnamed: 0', 'device_id', 'gender', 'age', 'group'], dtype='object')

##3 5 Information of the GAT

In [64]:
dt_gat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74645 entries, 0 to 74644
Data columns (total 5 columns):
Unnamed: 0    74645 non-null int64
device_id     74645 non-null int64
gender        74645 non-null object
age           74645 non-null int64
group         74645 non-null object
dtypes: int64(3), object(2)
memory usage: 2.8+ MB


##3 6 Description of the GAT

In [65]:
dt_gat.describe()

Unnamed: 0     device_id           age
count  74645.000000  7.464500e+04  74645.000000
mean   37322.000000 -7.491354e+14     31.410342
std    21548.299759  5.327150e+18      9.868735
min        0.000000 -9.223067e+18      1.000000
25%    18661.000000 -4.617367e+18     25.000000
50%    37322.000000 -1.841362e+16     29.000000
75%    55983.000000  4.636656e+18     36.000000
max    74644.000000  9.222849e+18     96.000000

In [66]:
dt_gat.describe(include="all")

Unnamed: 0     device_id gender           age   group
count   74645.000000  7.464500e+04  74645  74645.000000   74645
unique           NaN           NaN      2           NaN      12
top              NaN           NaN      M           NaN  M23-26
freq             NaN           NaN  47904           NaN    9605
mean    37322.000000 -7.491354e+14    NaN     31.410342     NaN
std     21548.299759  5.327150e+18    NaN      9.868735     NaN
min         0.000000 -9.223067e+18    NaN      1.000000     NaN
25%     18661.000000 -4.617367e+18    NaN     25.000000     NaN
50%     37322.000000 -1.841362e+16    NaN     29.000000     NaN
75%     55983.000000  4.636656e+18    NaN     36.000000     NaN
max     74644.000000  9.222849e+18    NaN     96.000000     NaN

##3 7 Counting Null of each Column

In [67]:
dt_gat.isnull().sum()

Unnamed: 0    0
device_id     0
gender        0
age           0
group         0
dtype: int64

# 4 Working on Phone_Brand_Device_Model<a name = "s13"></a>

##4 1 Read PhoneBrandDeviceModel Data to a Dataframe

In [0]:
dt_pbd = pd.read_csv('/content/data/phone_brand_device_model.csv')

##4 2 Shape of the PBD data

In [69]:
dt_pbd.shape

(87726, 4)

##4 3 Display first 10 rows of the PBD Data

In [70]:
dt_pbd.head(10)

Unnamed: 0            device_id phone_brand        device_model
0           0  1877775838486905855        vivo                 Y13
1           1 -3766087376657242966          小米                V183
2           2 -6238937574958215831        OPPO                 R7s
3           3  8973197758510677470          三星               A368t
4           4 -2015528097870762664          小米             红米Note2
5           5 -8896952452425303597          小米             红米Note3
6           6 -2634171391322875113          小米                S5.1
7           7 -2526386806972133373          三星  Galaxy Grand Prime
8           8  1427934937997164515          小米               MI 2S
9           9  2866484947180243791          三星       Galaxy Note 2

##4 4 Columns of the PBD Data

In [71]:
dt_pbd.columns

Index(['Unnamed: 0', 'device_id', 'phone_brand', 'device_model'], dtype='object')

##4 5 Information of the PBD data

In [72]:
dt_pbd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87726 entries, 0 to 87725
Data columns (total 4 columns):
Unnamed: 0      87726 non-null int64
device_id       87726 non-null int64
phone_brand     87726 non-null object
device_model    87726 non-null object
dtypes: int64(2), object(2)
memory usage: 2.7+ MB


##4 6 Description of the PBD data

In [73]:
dt_pbd.describe()

Unnamed: 0     device_id
count  87726.000000  8.772600e+04
mean   43862.500000 -5.212557e+15
std    25324.459195  5.330157e+18
min        0.000000 -9.223322e+18
25%    21931.250000 -4.623331e+18
50%    43862.500000 -3.109802e+16
75%    65793.750000  4.635194e+18
max    87725.000000  9.222849e+18

In [74]:
dt_pbd.describe(include="all")

Unnamed: 0     device_id phone_brand device_model
count   87726.000000  8.772600e+04       87726        87726
unique           NaN           NaN         116         1467
top              NaN           NaN          小米       红米note
freq             NaN           NaN       21645         3500
mean    43862.500000 -5.212557e+15         NaN          NaN
std     25324.459195  5.330157e+18         NaN          NaN
min         0.000000 -9.223322e+18         NaN          NaN
25%     21931.250000 -4.623331e+18         NaN          NaN
50%     43862.500000 -3.109802e+16         NaN          NaN
75%     65793.750000  4.635194e+18         NaN          NaN
max     87725.000000  9.222849e+18         NaN          NaN

##4 7 Counting Null of each Column

In [75]:
dt_pbd.isnull().sum()

Unnamed: 0      0
device_id       0
phone_brand     0
device_model    0
dtype: int64

# 5 Observations and Handling the data <a name = "s14"></a>



<br>Below data was observed from Events_data : 
<br>1. This table has 3252950 observations and 7 columns.
<br>2. Names of columns : 'event_id', 'device_id', 'timestamp', 'longitude', 'latitude', 'city', 'state'
<br>3. Datatypes : 
event_id     float64,  device_id    float64, timestamp    object, longitude    float64, latitude     float64, city         object, state        object
<br>4. 

| S.No 	| Column 	| Observation                  	|
|------	|--------------	|----------------------	|
| 1    	| device_id    	| It has 453 Null Values	|
| 2    	| Longitude          	| It has 423 Null Values|
| 4    	| Latitude          	| It has 423 Null Values|
| 4    	| State          	| It has 377 Null Values|


In [76]:
#Importing Pandas 
import pandas as pd
import numpy as np
import plotly.graph_objects as go
print("Import completed")

Import completed


In [77]:
# To connect to Google Drive : 
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [78]:
!ls drive/'My Drive'/1014capstone/
!mkdir -p data
!cp -R drive/'My Drive'/1014capstone/* data/

events_data.csv  gender_age_train.csv  phone_brand_device_model.csv


In [100]:
dt_evnts = pd.read_csv('/content/data/events_data.csv')
dt_evnts.head(5)

event_id     device_id            timestamp  ...  latitude   city  state
0       1.0  2.918269e+16  2016-05-01 00:55:25  ...  28.68161  Delhi  Delhi
1    7104.0  2.918269e+16  2016-05-02 09:37:02  ...  28.68161  Delhi  Delhi
2   29661.0  2.918269e+16  2016-05-04 00:56:04  ...  28.68161  Delhi  Delhi
3   33133.0  2.918269e+16  2016-05-06 05:01:15  ...  28.68161  Delhi  Delhi
4   38980.0  2.918269e+16  2016-05-06 09:55:04  ...  28.68161  Delhi  Delhi

[5 rows x 7 columns]

In [101]:
dt_evnts.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3252950 entries, 0 to 3252949
Data columns (total 7 columns):
event_id     float64
device_id    float64
timestamp    object
longitude    float64
latitude     float64
city         object
state        object
dtypes: float64(4), object(3)
memory usage: 173.7+ MB


In [102]:
dt_evnts.isnull().sum()

event_id       0
device_id    453
timestamp      0
longitude    423
latitude     423
city           0
state        377
dtype: int64

In [103]:
print("Count 1 : ", dt_evnts["device_id"].isnull().sum())
print("Count 2 : ", dt_evnts[dt_evnts["device_id"].isnull()].count())

Count 1 :  453
Count 2 :  event_id     453
device_id      0
timestamp    453
longitude    453
latitude     453
city         453
state        453
dtype: int64


In [104]:
dt_evnts[dt_evnts["device_id"].isnull()]

event_id  device_id            timestamp  ...   latitude     city    state
23808      72957.0        NaN  2016-05-03 06:10:14  ...  28.728888    Delhi    Delhi
23820      78897.0        NaN  2016-05-02 11:25:47  ...  28.728888    Delhi    Delhi
23831      94768.0        NaN  2016-05-02 17:39:12  ...  28.728888    Delhi    Delhi
23852     127677.0        NaN  2016-05-01 10:06:53  ...  28.728888    Delhi    Delhi
24093     442470.0        NaN  2016-05-07 09:03:18  ...  28.728888    Delhi    Delhi
...            ...        ...                  ...  ...        ...      ...      ...
3098746  1882296.0        NaN  2016-05-01 13:17:54  ...  21.194283  Bardoli  Gujarat
3098757  1952481.0        NaN  2016-05-05 00:53:42  ...  21.194283  Bardoli  Gujarat
3098801  2235564.0        NaN  2016-05-01 12:43:58  ...  21.194283  Bardoli  Gujarat
3098833  2452592.0        NaN  2016-05-02 13:34:20  ...  21.194283  Bardoli  Gujarat
3098901  3050759.0        NaN  2016-05-05 01:02:21  ...  21.194283  Bardoli  Gujarat

[453 rows x 7 columns]

## Remove the Null values and analyze the Data

In [0]:
dt_evnts1 = dt_evnts

In [106]:
dt_evnts1.isnull().sum()

event_id       0
device_id    453
timestamp      0
longitude    423
latitude     423
city           0
state        377
dtype: int64

In [0]:
dt_evnts1.dropna(inplace=True)

In [108]:
dt_evnts1.isnull().sum()

event_id     0
device_id    0
timestamp    0
longitude    0
latitude     0
city         0
state        0
dtype: int64

In [146]:
dt_evnts1.head(10)

event_id     device_id            timestamp  ...  latitude   city  state
0       1.0  2.918269e+16  2016-05-01 00:55:25  ...  28.68161  Delhi  Delhi
1    7104.0  2.918269e+16  2016-05-02 09:37:02  ...  28.68161  Delhi  Delhi
2   29661.0  2.918269e+16  2016-05-04 00:56:04  ...  28.68161  Delhi  Delhi
3   33133.0  2.918269e+16  2016-05-06 05:01:15  ...  28.68161  Delhi  Delhi
4   38980.0  2.918269e+16  2016-05-06 09:55:04  ...  28.68161  Delhi  Delhi
5   66398.0  2.918269e+16  2016-05-01 01:56:06  ...  28.68161  Delhi  Delhi
6   70894.0  2.918269e+16  2016-05-06 01:14:46  ...  28.68161  Delhi  Delhi
7   87185.0  2.918269e+16  2016-05-01 09:20:01  ...  28.68161  Delhi  Delhi
8   94644.0  2.918269e+16  2016-05-07 07:16:35  ...  28.68161  Delhi  Delhi
9  110018.0  2.918269e+16  2016-05-06 21:00:40  ...  28.68161  Delhi  Delhi

[10 rows x 7 columns]

In [206]:
dt_evnts1['device_id'].round(decimals=5) # not working as expected
# https://stackoverflow.com/questions/21137150/format-suppress-scientific-notation-from-python-pandas-aggregation-results -----> check this url to round the value
https://stackoverflow.com/questions/21137150/format-suppress-scientific-notation-from-python-pandas-aggregation-results

0          2.918269e+16
1          2.918269e+16
2          2.918269e+16
3          2.918269e+16
4          2.918269e+16
               ...     
3252945    6.057196e+18
3252946    8.855298e+18
3252947    1.734091e+18
3252948    7.886185e+18
3252949   -8.448990e+18
Name: device_id, Length: 3251697, dtype: float64

In [202]:
dt_evnts1.round(3)

event_id     device_id  ...         city       state
0              1.0  2.918269e+16  ...        Delhi       Delhi
1           7104.0  2.918269e+16  ...        Delhi       Delhi
2          29661.0  2.918269e+16  ...        Delhi       Delhi
3          33133.0  2.918269e+16  ...        Delhi       Delhi
4          38980.0  2.918269e+16  ...        Delhi       Delhi
...            ...           ...  ...          ...         ...
3252945  3241600.0  6.057196e+18  ...  Chitradurga   Karnataka
3252946  3241929.0  8.855298e+18  ...     Jangipur  WestBengal
3252947  3242387.0  1.734091e+18  ...     Khagaria       Bihar
3252948  3248517.0  7.886185e+18  ...        Bally  WestBengal
3252949  3251669.0 -8.448990e+18  ...    Bhavnagar     Gujarat

[3251697 rows x 7 columns]

In [0]:
# dt_evnts1.groupby('city','state')['device_id'].value_counts()
# dt_evnts1.groupby(['city','state','longitude','latitude'])['device_id'].value_counts()
# dt_evnts1.groupby(['longitude','latitude','city','state'])['device_id'].value_counts()

In [0]:
abc = dt_evnts1.groupby(['city','state','longitude','latitude'])['device_id'].value_counts() # working
#  abc = dt_evnts1.groupby(['city','state','longitude','latitude'])['device_id'] # working ---> we are not getting column names and values

In [198]:
type(abc)

pandas.core.series.Series

In [204]:
abc.head()

city    state   longitude  latitude   device_id    
Abohar  Punjab  74.217755  30.197155   7.829131e+18      6
                74.227885  30.181167  -6.793148e+18     15
                74.228694  30.163109   1.180390e+17     48
                74.231333  30.159260   7.162956e+18    255
                74.236340  30.235750   4.772855e+18      2
Name: device_id, dtype: int64

In [0]:
# #Below all are good

# dt_evnts1.groupby(['city','state','longitude','latitude'])['device_id'].head(5)
# dt_evnts1.groupby(['city','state'])['device_id'].value_counts()
# dt_evnts1.groupby(['city','state'])['device_id'].head(3)
# dt_evnts1.groupby(['city','state'])['device_id']
# dt_evnts1.groupby(['city','state'])['city'].value_counts()
# dt_evnts1[(dt_evnts1['state']=="Maharashtra")].count()   
# dt_evnts1[(dt_evnts1['city']=="Abohar") & (dt_evnts1['state']=="Punjab")].count()  
# dt_evnts1[(dt_evnts1['city']=="Adoni") & (dt_evnts1['state']=="AndhraPradesh")].count()  
# dt_evnts1[dt_evnts1['state']=="Maharashtra"].count() 

#Not working
# dt_evnts1[(dt_evnts1['city']=="Pune") or (dt_evnts1['state']=="Maharashtra")] wrong
# dt_evnts1[(dt_evnts1['city']=="Pune") & (dt_evnts1['state']=="Maharashtra") & (dt_evnts1['longitude']=="12.567000")].count()  # not working
# dt_evnts1[(dt_evnts1['city']=="Abohar") & (dt_evnts1['state']=="Punjab") & (dt_evnts1['longitude']==74.217755)].count()   # not working

dt_evnts1[(dt_evnts1['longitude']==74.217755)].count()   # not working